renormalizer/mps/gs.py

```diff
@@ -107,12 +107,21 @@ def optimize_mps(mps: Mps, mpo: Mpo, omega: float = None) -> Tuple[List, Mps]:
     # determines the index of active site of the returned mps
     opt_e_idx: int = None
     res_mps: Union[Mps, List[Mps]] = None
-    for isweep, (mmax, percent) in enumerate(mps.optimize_config.procedure):
+    for isweep, (compress_config, percent) in enumerate(mps.optimize_config.procedure):
         logger.debug(f"isweep: {isweep}")
-        logger.debug(f"mmax, percent: {mmax}, {percent}")
+        
+        if isinstance(compress_config, CompressConfig):
+            mps.compress_config = compress_config
```

Comment:
What about the original `mps.compress_config`? Should we record it and then setting it back after the optimization?

Reply:
We control the bond dimension in gs.py with mmax and the original mps.compress_config is not used. I guess it is the default one. So is it necessary to set it back?

Reply:
The user may have set this value for future time evolution. pseudo-code:
```
mps.compress_config = ...
mps.optimize_config = ...
mps = optimize_mps(mps, mpo, procedure)
mps = excitation_mpo @ mps
mps = mps.evolve_with_rk(tau)
```

Reply:
I got it.

Reply:
solved.